README.md

```diff
@@ -32,47 +32,70 @@ This must only be done once per component.
 
-### Using a client
-
-Clients are used like regular libraries. The simples client is the `Log` client that provides standard logging facilities.
+The simplest interfaces are used like regular libraries. An example is the `Log` client that provides standard logging facilities.
 A hello world with the component described above looks as follows
 
 ```Ada
-with Cai.Log;
-with Cai.Log.Client;
+with Componolit.Gneiss.Log;
+with Componolit.Gneiss.Log.Client;
 
 package body Component is
 
-   procedure Initialize (Cap : Cai.Types.Capability)
+   Log : Componolit.Gneiss.Log.Client_Session;
+
+   procedure Construct (Cap : Componolit.Gneiss.Types.Capability)
    is
-      Log : Cai.Log.Client_Session;
    begin
-      Cai.Log.Client.Initialize (Log, Cap, "Hello_World");
-      Cai.Log.Client.Info (Log, "Hello World!");
-      Cai.Log.Client.Warning (Log, "Hello World!");
-      Cai.Log.Client.Error (Log, "Hello World!");
-      Cai.Log.Client.Finalize (Log);
-   end Initialize;
+      if not Componolit.Gneiss.Log.Initialized (Log) then
+         Componolit.Gneiss.Log.Client.Initialize (Log, Cap, "Hello_World");
+      end if;
+      if Componolit.Gneiss.Log.Initialized (Log) then 
+         Componolit.Gneiss.Log.Client.Info (Log, "Hello World!");
+         Componolit.Gneiss.Log.Client.Warning (Log, "Hello World!");
+         Componolit.Gneiss.Log.Client.Error (Log, "Hello World!");
+        Main.Vacate (Cap, Main.Success);
+      else
+        Main.Vacate (Cap, Main.Failure);
+      end if;
+   end Construct;
+
+   procedure Destruct
+   is
+   begin
+      if Componolit.Gneiss.Log.Initialized (Log) then
+         Componolit.Gneiss.Log.Finalize (Log);
+      end if;
+   end Destruct;
    
 end Component;
 ```
 
-This component opens a `Log` client session with its provided capability and then prints "Hello World!" as info, warning and error. Since there is only a initialize procedure that will only be called once the `Log` session is closed afterwards.
-This produces partially different outputs on different platforms:
+In `Construct` the component checks if the log session `Log` is already initialized and initialized it if that is not the case.
+Initializing a session that is already initialized could lead to double initialization and crashes or memory leaks on the platform.
+Since the initialization can fail it checks again if it succeeded.
+If this is the case it prints "Hello World!" as an info, warning and error message.
+The component then calls `Vacate` to tell the platform that it has finished its work and can be terminated.
+`Vacate` does not immediately kill the component but tell the platform that it can be stopped now. The current method will still return normally
+and there is no guarantee not to be called again.
+When the platform at some point decides to terminate the component it will call `Destruct`.
```

Comment:
What is it good for (beyond the logging example)?

Reply:
Addressed by https://github.com/Componolit/gneiss/pull/92/commits/db86420560310e1980fe1d0a46a73f93c1080751.